Package/gulpfile.js

```diff
@@ -0,0 +1,117 @@
+  await exec('mkdir -p iOS/Build');
+  await exec('cmake -G Xcode -DCMAKE_TOOLCHAIN_FILE=../../../Apps/Playground/node_modules/@babylonjs/react-native/submodules/BabylonNative/Dependencies/ios-cmake/ios.toolchain.cmake -DPLATFORM=OS64COMBINED -DENABLE_ARC=0 -DDEPLOYMENT_TARGET=12 -DENABLE_GLSLANG_BINARIES=OFF -DSPIRV_CROSS_CLI=OFF ..', {cwd: 'iOS/Build'});
+};
+
+const buildIphoneOS = async () => {
+  await exec('xcodebuild -sdk iphoneos -configuration Release -project ReactNativeBabylon.xcodeproj -scheme BabylonNative build CODE_SIGNING_ALLOWED=NO', {cwd: 'iOS/Build'});
+};
+
+const buildIphoneSimulator = async () => {
+  await exec('xcodebuild -sdk iphonesimulator -configuration Release -project ReactNativeBabylon.xcodeproj -scheme BabylonNative build CODE_SIGNING_ALLOWED=NO', {cwd: 'iOS/Build'});
+};
+
+const buildIOS = gulp.series(makeXCodeProj, buildIphoneOS, buildIphoneSimulator);
+
+const buildAndroid = async () => {
+  await exec('./gradlew babylonjs_react-native:assembleRelease', {cwd: '../Apps/Playground/android'});
+};
+
+const copyCommonFiles = async () => {
+  return  gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/package.json')
+    .pipe(gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/README.md'))
+    .pipe(gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/*.ts*'))
+    .pipe(gulp.src('react-native-babylon.podspec'))
+    .pipe(gulp.dest('Assembled'));
+};
+
+const copyIOSFiles = async () => {
+  return  gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/ios/*.h')
+    .pipe(gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/ios/*.mm'))
+    // This xcodeproj is garbage that we don't need in the package, but `pod install` ignores the package if it doesn't contain at least one xcodeproj. 🤷🏼‍♂️
+    .pipe(gulp.src('iOS/Build/ReactNativeBabylon.xcodeproj**/**/*'))
+    .pipe(gulp.dest('Assembled/ios'));
+};
+
+const createIOSUniversalLibs = async () => {
+  await exec('mkdir -p Assembled/ios/libs');
+  const libs = await readdirAsync('iOS/Build/Release-iphoneos');
+  await Promise.all(libs.map(lib => exec(`lipo -create iOS/Build/Release-iphoneos/${lib} iOS/Build/Release-iphonesimulator/${lib} -output Assembled/ios/libs/${lib}`)));
+};
+
+const copyAndroidFiles = async () => {
+  await new Promise(resolve => {
+          gulp.src('Android/build.gradle')
+    .pipe(gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/android/src**/main/AndroidManifest.xml'))
+    .pipe(gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/android/src**/main/java/**/*'))
+    .pipe(gulp.dest('Assembled/android'))
+    .on('end', resolve);
+  });
+
+  await new Promise(resolve => {
+          gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/android/build/intermediates/library_and_local_jars_jni/release/jni/**/*')
+    .pipe(gulp.dest('Assembled/android/src/main/jniLibs/'))
+    .on('end', resolve);
+  });
```

Comment:
Is using `new Promise` really necessary here? Seems like you can use `gulp.parallel` like below?

Reply:
Well, if i use `gulp.parallel`, then Gulp treats these as individual atomic "tasks." I think it makes more sense to have these grouped together in a single "logical task."

Reply:
I mean use gulp.parallel on the "tasks" returned by each section of the gulp.pipe.pipe:

```suggestion
  gulp.parallel(
          gulp.src('Android/build.gradle')
    .pipe(gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/android/src**/main/AndroidManifest.xml'))
    .pipe(gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/android/src**/main/java/**/*'))
    .pipe(gulp.dest('Assembled/android')),
          gulp.src('../Apps/Playground/node_modules/@babylonjs/react-native/android/build/intermediates/library_and_local_jars_jni/release/jni/**/*')
    .pipe(gulp.dest('Assembled/android/src/main/jniLibs/'))
  );
```

Does this work?

Reply:
No, that will create two nodes in the task hierarchy named "anonymous function" (or something like that). Talked offline, leaving as is.